mobile/packages/configuration/lib/base_configuration.dart

```diff
@@ -85,11 +85,50 @@ class BaseConfiguration {
   }
 
   Future<void> resetSecureStorage() async {
-    // Delete all keys except preserved ones
-    final allKeys = await _secureStorage.readAll();
-    for (final key in allKeys.keys) {
-      if (!preservedKeys.contains(key)) {
-        await _secureStorage.delete(key: key);
+    // Try to preserve offline key if it exists
+    String? offlineKey;
+    try {
+      offlineKey = await _secureStorage.read(key: offlineAuthSecretKey);
+    } catch (e) {
+      _logger.warning('Failed to read offline key: $e');
+      // Continue with reset even if we can't read the offline key
+    }
+
+    try {
+      // Try selective deletion first (preserving specified keys)
+      final allKeys = await _secureStorage.readAll();
+      for (final key in allKeys.keys) {
+        if (!preservedKeys.contains(key)) {
+          await _secureStorage.delete(key: key);
+        }
+      }
+    } catch (e) {
+      // If readAll or delete operations fail (e.g., BadPaddingException),
+      // fall back to deleting everything
+      _logger.warning(
+        'Error during selective deletion, falling back to deleteAll: $e',
+      );
+      try {
+        await _secureStorage.deleteAll();
+        // Try to restore the offline key if we had it
+        if (offlineKey != null) {
+          try {
```

Comment:
Let's only clear if offlineKey is missing? write might also have chance of failing.
Ideally, we should never run into this catch block in the offline mode.